crates/bindings-csharp/BSATN.Runtime/BSATN/Int256.cs

```diff
@@ -0,0 +1,134 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+
+namespace SpacetimeDB
+{
+    /// <summary>Represents a 256-bit signed integer.</summary>
+    [StructLayout(LayoutKind.Sequential)]
+    public readonly struct Int256 : IEquatable<Int256>
+    {
+        internal const int Size = 32;
+
+#if BIGENDIAN
+        private readonly UInt128 _upper;
+        private readonly UInt128 _lower;
+#else
+        private readonly UInt128 _lower;
+        private readonly UInt128 _upper;
+#endif
+
+        /// <summary>Initializes a new instance of the <see cref="Int256" /> struct.</summary>
+        /// <param name="upper">The upper 128-bits of the 256-bit value.</param>
+        /// <param name="lower">The lower 128-bits of the 256-bit value.</param>
+        public Int256(UInt128 upper, UInt128 lower)
+        {
+            _lower = lower;
+            _upper = upper;
+        }
+
+        internal UInt128 Lower => _lower;
+
+        internal UInt128 Upper => _upper;
+
+        /// <inheritdoc cref="IComparable.CompareTo(object)" />
+        public int CompareTo(object? value)
```

Comment:
I think these all implementations could be significantly simplified if you declared this type as a wrapper around `(UInt128 Upper, UInt128 Lower)` tuple field instead. In C# tuples automatically implement `CompareTo`, equality and `GetHashCode`, so you'd be able to simply forward all those methods to the field instead of reimplementing them from scratch.

Reply:
Signed types might need a bit more care though, not sure.